src/walk.rs

```diff
@@ -248,6 +248,16 @@ pub fn scan(path_vec: &[PathBuf], pattern: Arc<Regex>, config: Arc<FdOptions>) {
                 }
             }
 
+            // Filter out unwanted sizes if it is a file and we have been given size constraints.
+            if entry_path.is_file() && config.size_constraints.len() > 0 {
```

Comment:
It might be premature optimization, but I think we should probably swap these two conditions (and verify by a benchmark). The way it is written right now would lead to a syscall even when there are no size constraints.

Reply:
Yeah, it would be better that way